File: revature_workspace/JDBCBank/src/main/java/com/revature/beans/Account.java
package com.revature.beans;

import com.revature.Exceptions.ZeroBalanceException;

public abstract class Account {
	public float initialBalance;
	public AccountType accountType;
	public int userId;
	public Account(){}
	public Account (float initialBalance, int userID) throws ZeroBalanceException
	{
		try{
			if(initialBalance<1)
			{ 
				throw new ZeroBalanceException("Your initial balance cannot be less than zero.");
			
			}
			else{
				this.initialBalance=initialBalance;
				this.userId=userID;
			
			}
			}
			catch(ZeroBalanceException e)
			{
				e.printStackTrace();
			}

	}
	public Account(float initialBalance,
			AccountType accountType,int userID)
	{
		
		this.initialBalance=initialBalance;
		this.accountType=accountType;
		this.userId=userID;
	}
	
	 public abstract void deposit(float funds);
	 public abstract void withdraw(float withdrawal);
	 public abstract float getBalance();
	 public abstract void displayBalance();
	 public abstract void closeAccount();
	 public abstract AccountType getAccountType();

}

File: typescript-bank/account.js
"use strict";
exports.__esModule = true;
var Account = /** @class */ (function () {
    function Account(balance, type) {
        this.balance = balance;
        this.type = type;
    }
    Account.prototype.withdraw = function (amount) {
        this.balance = this.balance - amount;
    };
    Account.prototype.deposit = function (amount) {
        this.balance = this.balance + amount;
    };
    return Account;
}());
exports.Account = Account;
